File: Models/Book.swift
import Foundation

struct Book: Identifiable, Codable, Equatable {
    let id: String
    let title: String
    let coverURL: String
    let ageMin: Int
    let ageMax: Int
    let tags: [String]
    let popularityScore: Double
    let themeKeywords: [String]
    let isEditorPick: Bool
    let pages: [PageModel]
}

extension Book {
    func ageFits(_ age: Int) -> Double {
        if age < ageMin { return 0 }
        if age > ageMax { return 0 }
        let span = max(ageMax - ageMin, 1)
        let pos = Double(age - ageMin) / Double(span)
        return 1 - abs(pos - 0.5) * 2
    }
}

File: server/internal/models/book.go
package models

type Page struct {
    ID        string  `json:"id"`
    Index     int     `json:"index"`
    ImageURL  string  `json:"imageURL"`
    Duration  *int    `json:"duration_hint,omitempty"`
}

type Book struct {
    ID              string   `json:"id"`
    Title           string   `json:"title"`
    CoverURL        string   `json:"coverURL"`
    AgeMin          int      `json:"ageMin"`
    AgeMax          int      `json:"ageMax"`
    Tags            []string `json:"tags"`
    PopularityScore float64  `json:"popularityScore"`
    ThemeKeywords   []string `json:"themeKeywords"`
    IsEditorPick    bool     `json:"isEditorPick"`
    Pages           []Page   `json:"pages"`
    Status          string   `json:"status"`
}